tests/federation_room_invite_test.go

```diff
@@ -0,0 +1,83 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/matrix-org/gomatrixserverlib"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/federation"
+)
+
+// This test ensures that invite rejections are correctly sent out over federation.
+//
+// We start with two users in a room - alice@hs1, and 'delia' on the Complement test server.
+// alice sends an invite to charlie@hs2, which he rejects.
+// We check that delia sees the rejection.
+//
+func TestFederationRejectInvite(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintFederationTwoLocalOneRemote)
+	defer deployment.Destroy(t)
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	charlie := deployment.Client(t, "hs2", "@charlie:hs2")
+
+	// we'll awaken this Waiter when we receive a membership event for Charlie
+	var waiter *Waiter
+
+	srv := federation.NewServer(t, deployment,
+		federation.HandleKeyRequests(),
+		federation.HandleTransactionRequests(func(ev *gomatrixserverlib.Event) {
+			sk := "<nil>"
+			if ev.StateKey() != nil {
+				sk = *ev.StateKey()
+			}
+			t.Logf("Received PDU %s/%s", ev.Type(), sk)
+			if waiter != nil && ev.Type() == "m.room.member" && ev.StateKey() != nil && *ev.StateKey() == charlie.UserID {
```

Comment:
You can use `ev.StateKeyEquals(charlie.UserID)` which does the state key nil check for you: https://pkg.go.dev/github.com/matrix-org/gomatrixserverlib#Event.StateKeyEquals